fitbenchmarking/hessian/analytic_hessian.py

```diff
@@ -32,20 +32,36 @@ def eval(self, params, **kwargs):
         """
 
         x = kwargs.get("x", self.problem.data_x)
-        # y = kwargs.get("y", self.problem.data_y)
+        y = kwargs.get("y", self.problem.data_y)
         e = kwargs.get("e", self.problem.data_e)
+        grad2_r = -self.problem.hessian(x, params)
         J = self.jacobian.eval(params, **kwargs)
-        rx = self.cached_func_values(self.cost_func.cache_rx,
-                                     self.cost_func.eval_r,
-                                     params,
-                                     **kwargs)
-        grad2_r = self.problem.hessian(x, params)
         if self.problem.options.cost_func_type == "weighted_nlls":
             # scales the Hessian by the weights
             for i in range(len(e)):
                 grad2_r[:, :, i] = grad2_r[:, :, i] / e[i]
+        if self.problem.options.cost_func_type == "hellinger_nlls":
+            model_eval = self.problem.eval_model(params, x=x)
+            for i, (f, j) in enumerate(zip(model_eval, J)):
+                j = np.array([j])
+                grad2_r[:, :, i] = 1/2 * f**(-1/2) * grad2_r[:, :, i] \
+                    + f**(-1/2) * np.matmul(j.T, j)
+        if self.problem.options.cost_func_type == "poisson":
+            model_eval = self.problem.eval_model(params, x=x)
+            for i, (f, j) in enumerate(zip(model_eval, J)):
+                j = np.array([j])
+                grad2_r[:, :, i] = (-grad2_r[:, :, i] * (1-y[i]/f)
+                                    + (y[i]/f**2)*(1/(1-y[i]/f)**2)
```

Comment:
I think this is right but the f's cancel? I doubt it will change performance but it will reduce the number of operations